src/Framework/Framework/ViewModel/Serialization/ViewModelSerializationMap.cs

```diff
@@ -246,16 +301,30 @@ public ReaderDelegate CreateReaderFactory()
             // encryptedValuesReader.AssertEnd();
             block.Add(Expression.Call(encryptedValuesReader, nameof(EncryptedValuesReader.AssertEnd), Type.EmptyTypes));
 
+            // call the constructor
+            if (hasConstructorProperties)
+            {
+                block.Add(Assign(value, constructorCall));
+            }
+
+            var setProperties = Expression.Block(
+                Properties
+                    .Zip(propertyVars, (p, v) =>
+                        p is { PropertyInfo.SetMethod: not null, ConstructorParameter: null, TransferToServer: true }
+                            ? Expression.Assign(Expression.Property(value, p.PropertyInfo), v)
+                            : null)
+                    .Where(e => e != null)!
+            );
+            block.Add(setProperties);
+
             // return value
-            block.Add(value);
+            block.Add(Convert(value, typeof(object)));
 
             // build the lambda expression
             var ex = Expression.Lambda<ReaderDelegate>(
-                Expression.Convert(
-                    Expression.Block(Type, new[] { value, currentProperty, readerTmp }, block),
-                    typeof(object)).OptimizeConstants(),
-                reader, serializer, valueParam, encryptedValuesReader);
-            return ex.CompileFast(flags: CompilerFlags.ThrowOnNotSupportedExpression);
+                Expression.Block(typeof(object), new[] { value, currentProperty, readerTmp }.Concat(propertyVars), block).OptimizeConstants(),
+                reader, serializer, valueParam, encryptedValuesReader, servicesParameter);
+            return ex.CompileFast(flags: CompilerFlags.ThrowOnNotSupportedExpression | CompilerFlags.EnableDelegateDebugInfo);
```

Comment:
Please remove `CompilerFlags.EnableDelegateDebugInfo` or leave it only for debug compilation.